pimd/pim_mroute.c

```diff
@@ -55,18 +55,34 @@ int pim_mroute_msg_nocache(int fd, struct interface *ifp, const kernmsg *msg)
 	struct pim_rpf *rpg;
 	pim_sgaddr sg;
 
-	rpg = pim_ifp ? RP(pim_ifp->pim, msg->msg_im_dst) : NULL;
+	memset(&sg, 0, sizeof(sg));
+	sg.src = msg->msg_im_src;
+	sg.grp = msg->msg_im_dst;
+
+	if (!pim_ifp) {
+		if (PIM_DEBUG_MROUTE)
```

Comment:
I put this as a detail because this *absolutely* filled up the log system when we start receiving packets at a very very fast rate.  I don't disagree that this would help debug, but we need something better here.

Reply:
Do you know which of the 3 conditions was causing this? There was no way previously to distinguish which caused it from the log message alone, so you need context…

Reply:
For those of you following along at home, David and I spoke about this in slack.

The scenario where we kept seeing this type of behavior:  Router is running both say bgp and pim and the router enters into a problem state and is rebooted.  In this case PIM comes back up pretty quickly and starts receiving the _nocache callbacks at line rate for some stream that was working prior to the crash/reboot/whatever.  BGP has not come back and the RPF check here can't work until BGP is up enough to actually receive and install the route.  This will quickly fill up the route with large amounts of data